src/roles/pulp/tasks/main.yaml

```
@@ -128,41 +130,34 @@
     name: pulpcore-manager-migrate
     image: "{{ pulp_api_image }}"
     command: pulpcore-manager migrate --noinput
+    detach: false
     network: host
     secrets:
       - 'pulp-symmetric-key,type=mount,target=/etc/pulp/certs/database_fields.symmetric.key'
       - 'pulp-db-password,type=env,target=PULP_DATABASES__default__PASSWORD'
     env: "{{ pulp_settings_database_env }}"
 
-- name: Start the Pulp API services
+- name: Start Pulp services
   ansible.builtin.systemd:
-    name: pulp-api
-    enabled: true
-    state: started
-
-- name: Wait for Pulp API service to be accessible
-  ansible.builtin.wait_for:
-    host: "{{ ansible_fqdn}}"
-    port: 24817
-    timeout: 300
-
-- name: Start the Pulp Content services
-  ansible.builtin.systemd:
-    name: pulp-content
-    enabled: true
-    state: started
-
-- name: Wait for Pulp Content service to be accessible
-  ansible.builtin.wait_for:
-    host: "{{ ansible_fqdn }}"
-    port: 24816
-    timeout: 600
-
-- name: Start the Pulp Worker service
-  ansible.builtin.systemd:
-    name: pulp-worker
+    name: "{{ item }}"
     enabled: true
     state: started
+  async: 3
+  poll: 0
+  loop:
+    - pulp-api
+    - pulp-content
+    - pulp-worker
+  register: pulp_services
+
+- name: Wait for Pulp services
+  async_status:
```

seems my suggestion offended the :policeman: and this needs to be
```suggestion
  ansible.builtin.async_status:
```